packages/key4hep-stack/package.py

```diff
@@ -232,4 +241,38 @@ class Key4hepStack(BundlePackage):
               "which are therefore not supported." \
               "See https://root-forum.cern.ch/t/devtoolset-gcc-toolset-compatibility/38286")
 
+    def install(self, spec, prefix):
+      specs = [spec]
+      with spack.store.db.read_transaction():
+               specs = [dep for _spec in specs
+                        for dep in
+                        _spec.traverse( order='post')]
+              
+               if "+load_gcc" in spec:
+                 gcc_specs = [spack.cmd.disambiguate_spec("gcc", None, first=True)]
```

Comment:
Will this always pick up the gcc version that has been used to build the `key4hep-stack`?

Reply:
No, it's true that this could be improved by parsing the spec. Having several gcc installations around is not a likely scenario though - this is really geared towards automating the workflow on build machines.